yucca/training/data_loading/YuccaDataset.py

```diff
@@ -11,22 +12,21 @@
 class YuccaTrainDataset(torch.utils.data.Dataset):
     def __init__(
         self,
-        preprocessed_data_dir: list,
+        samples: list,
         patch_size: list | tuple,
         keep_in_ram: Union[bool, None] = None,
         label_dtype: Optional[Union[int, float]] = None,
         composed_transforms=None,
-        task_type: Literal["classification", "segmentation", "unsupervised"] = "segmentation",
+        task_type: Literal["classification", "segmentation", "unsupervised", "contrastive"] = "segmentation",
     ):
-        self.all_cases = preprocessed_data_dir
+        self.all_cases = samples
         self.composed_transforms = composed_transforms
         self.patch_size = patch_size
         self.task_type = task_type
-        if label_dtype is None:
-            if self.task_type in ["segmentation", "classification"]:
-                self.label_dtype = torch.int32
-            if self.task_type == "unsupervised":
-                self.label_dtype = torch.float32
+        if label_dtype is None and self.task_type in ["segmentation", "classification"]:
+            self.label_dtype = torch.int32
+        else:
+            self.label_dtype = label_dtype
```

Comment:
@Sllambias these changes seem to work and makes a lot more sense to me, but might be wrong, so please let me know if i have misunderstood something! (the old change used float for tasks that did not use labels and did not pass on the `label_dtype` param)

Reply:
I would like to have a one line comment here, why could it happen that label_dtype is None in either the segmentation or classification setting?